src/osd/OSD.cc

```diff
@@ -7108,9 +7112,10 @@ bool OSD::compat_must_dispatch_immediately(PG *pg)
        ++i) {
     if (i->osd == whoami || i->osd == CRUSH_ITEM_NONE)
       continue;
+    uint64_t features;
     ConnectionRef conn =
-      service.get_con_osd_cluster(i->osd, pg->get_osdmap()->get_epoch());
-    if (conn && !conn->has_feature(CEPH_FEATURE_INDEP_PG_MAP)) {
+      service.get_con_osd_cluster(i->osd, pg->get_osdmap()->get_epoch(), &features);
+    if (!(features & CEPH_FEATURE_INDEP_PG_MAP)) {
```

Comment:
Is conn actually used here?  Perhaps we want to just look at OSDMap directly for the feature instead of going through the connection helper